### ISSUE TITLE
npm run inside scripts does not have access to private packages

### ISSUE THREAD

**Issue by Swatinem** [yarnpkg/yarn#5427]
**Do you want to request a *feature* or report a *bug*?**

A bug

**What is the current behavior?**

When I run `npm` inside a `script`, it does not have access to private repositories.

**If the current behavior is a bug, please provide the steps to reproduce.**

I have the following `package.json` (where `XXX` is any private package not accessible for unauthorized npm users):
```
{
    "scripts": {
        "foo": "npm show XXX versions"
    }
}
```

When I run `npm run foo`, the subcommand executes correctly

When I run `yarn run foo`, the subcommand fails:
```
yarn run v1.3.2
warning package.json: No license field
$ npm show XXX versions
npm ERR! code E404
npm ERR! 404 Not found : XXX
npm ERR! 404 
npm ERR! 404  'XXX' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     XXX
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**What is the expected behavior?**

Things should work the same, no matter if I run the script via `npm` or `yarn`.

**Please mention your node.js, yarn and operating system version.**

up to date Archlinux
```
yarn -v && npm -v && node -v
1.3.2
5.6.0
v9.6.1
```

**Comment by rally25rs** [comment]
Is this to a private npm package, or a custom registry? How do you provide your npm credentials for your private repo? `npm login`? `.npmrc` file? Thanks for any additional info.

**Comment by Swatinem** [comment]
It is a private package on the official npm registry.
I did a `npm login` quite some time ago, now I have the `authToken` defined in my `.npmrc` file.

Maybe something else that might help you:
The above example is a simplified case which fails, in our case we use `yarn run` to run a custom automation command which itself runs `npm` as a child_process.
I was able to work around this issue by removing every `npmXXX` environment variable that is passed to the child_process.
